pkg/datastore/caveat.go

```diff
@@ -11,8 +11,9 @@ type CaveatReader interface {
 	// ReadCaveatByName returns a caveat with the provided name
 	ReadCaveatByName(ctx context.Context, name string) (*core.CaveatDefinition, Revision, error)
 
-	// ListCaveats returns all caveats stored in the system
-	ListCaveats(ctx context.Context) ([]*core.CaveatDefinition, error)
+	// ListCaveats returns all caveats stored in the system. If caveatNames are provided
+	// the result will be filtered to the provided caveat names
+	ListCaveats(ctx context.Context, caveatNames ...string) ([]*core.CaveatDefinition, error)
```

Comment:
name the parameter `optionalCaveatNames` or `caveatNamesForFiltering` or somesuch